modal/image.py

```diff
@@ -264,6 +263,45 @@ class DockerfileSpec:
     context_files: Dict[str, str]
 
 
+async def _image_await_build_result(image_id: str, client: _Client) -> api_pb2.ImageJoinStreamingResponse:
```

Comment:
Mostly moved as is from below, to make code at call site more readable after branching